File: RS232/python/examples/main.py
#!/usr/bin/python
# -*- coding:utf-8 -*-
import serial
import os
import sys
import logging

logging.basicConfig(level=logging.INFO)
libdir = os.path.join(os.path.dirname(os.path.dirname(os.path.realpath(__file__))), 'lib')
if os.path.exists(libdir):
    sys.path.append(libdir)
    
import RPi.GPIO as GPIO
import time
from waveshare_UART import config

ser = config.config(dev = "/dev/ttyAMA1",Baudrate = 115200)
data = ''

print("A carriage return and line feed need to be added to the end of the sending string\r\n");
ser.Uart_SendString('Compute Module PoE 4G Board\r\n')
    
try:
    while(1):
        data_t = ser.Uart_ReceiveByte()
        data += str(data_t)
        if(data_t == '\n'):
            print(data)
            ser.Uart_SendString(data)
            data = ''
            
except KeyboardInterrupt:    
    logging.info("ctrl + c:")
    exit()


     
     
     
     

File: RS232/c/examples/main.c
#include <stdio.h>      //printf()
#include <stdlib.h>     //exit()
#include <signal.h>
 #include <stdio.h>
 #include <string.h>
#include "DEV_Config.h"

void  Handler(int signo)
{
    //System Exit
    printf("\r\nHandler:Program stop\r\n"); 
    DEV_ModuleExit();
    exit(0);
}

int main(int argc, char **argv)
{
    // Exception handling:ctrl + c
    signal(SIGINT, Handler);
    
    if (DEV_ModuleInit()==1)return 1;
    
    DEV_UART_Init("/dev/ttyAMA1");
    UART_Set_Baudrate(115200);
    UBYTE pData[1000]={0};
    int i=0,j=0;
    printf("A carriage return and line feed need to be added to the end of the sending string\r\n");
    UART_Write_nByte("Compute Module PoE 4G Board\r\n", 29);
    while(1){
        pData[i] = UART_Read_Byte();
        //printf("%c\r\n",pData[i]);
        if(pData[i] == '\n'|| pData[i] == '\r'){
            UART_Write_nByte(pData, i+1);
            printf("发送：%s\r\n",pData);
            memset(pData, 0, sizeof pData);   
            i=0;
        }else if(i>1000){
            i=0;
        }else{
            i++;
        }
    }
    
    DEV_ModuleExit();
    return 0; 
}
